src/main/java/org/sonarlint/intellij/util/VirtualFileUtils.kt

```diff
@@ -28,7 +28,7 @@ import java.net.URISyntaxException
 object VirtualFileUtils {
     fun toURI(file: VirtualFile): URI? {
         return try {
-            URI(file.url.replace(" ", "%20"))
+            URI("${file.fileSystem.protocol}:///${file.path}".replace(" ", "%20"))
```

Comment:
My concern is that this won't work with remote files. But maybe we already don't support this case with SLI?
Also I think you could at least print a log in the catch below

Reply:
You could maybe check if the `file.isInLocalFileSystem()`? If it's not you will try to build a local URI that maybe does not exist

Reply:
We were effectively only computing URI for file in local file system before SLOOP